src/main/scala/com/github/gvolpe/http4s/server/service/FileService.scala

```diff
@@ -34,12 +36,12 @@ class FileService[F[_]](implicit F: Effect[F], S: StreamUtils[F]) {
     }
   }
 
-  def store(part: Part[F]): Stream[F, Unit] =
+  def store(part: Part[F])(implicit cs: ContextShift[F]): Stream[F, Unit] =
     for {
       home      <- S.evalF(sys.env.getOrElse("HOME", "/tmp"))
       filename  <- S.evalF(part.filename.getOrElse("sample"))
       path      <- S.evalF(Paths.get(s"$home/$filename"))
-      _         <- part.body to fs2.io.file.writeAll(path)
+      _         <- part.body to fs2.io.file.writeAll(path, ExecutionContext.global)
```

Comment:
There's something in that line that doesn't work.

It breaks the multipart upload endpoint.

Reply:
The error could come from:

```scala
root 23:10:18.525 [scala-execution-context-global-10] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Websocket key: None
```

Here's the full trace:
```scala
root 23:10:15.515 [blaze-selector-0-0] DEBUG org.http4s.blaze.channel.nio1.SelectorLoop - Channel initialized.
root 23:10:15.536 [scala-execution-context-global-10] INFO org.http4s.server.blaze.BlazeServerBuilder - http4s v0.20.0-M4 on blaze v0.14.0-M11 started at http://[0:0:0:0:0:0:0:0]:8080/
root 23:10:17.444 [blaze-selector-0-1] DEBUG org.http4s.blaze.channel.nio1.SelectorLoop - Channel initialized.
root 23:10:17.445 [blaze-selector-0-1] DEBUG org.http4s.blaze.channel.nio1.NIO1HeadStage - Starting up.
root 23:10:17.445 [blaze-selector-0-1] DEBUG org.http4s.blaze.channel.nio1.NIO1HeadStage - Stage NIO1HeadStage sending inbound command: Connected
root 23:10:17.445 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Starting HTTP pipeline
root 23:10:17.448 [blaze-selector-0-1] DEBUG org.http4s.blazecore.IdleTimeoutStage - Starting idle timeout stage with timeout of 30000 ms
root 23:10:17.465 [blaze-selector-0-1] INFO org.http4s.blaze.channel.nio1.NIO1SocketServerGroup - Accepted connection from /0:0:0:0:0:0:0:1:58548
root 23:10:17.614 [scala-execution-context-global-10] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Parse result: None, content complete: false
root 23:10:18.412 [scala-execution-context-global-10] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Parse result: Some(java.nio.HeapByteBufferR[pos=0 lim=170 cap=170]), content complete: false
root 23:10:18.488 [scala-execution-context-global-10] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Parse result: None, content complete: false
root 23:10:18.489 [scala-execution-context-global-10] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Parse result: Some(java.nio.HeapByteBufferR[pos=0 lim=9653 cap=9653]), content complete: true
root 23:10:18.499 [scala-execution-context-global-10] ERROR com.github.gvolpe.http4s.server.endpoints.MultipartHttpEndpoint - HUUUUUUUUUUUUUGE 00
root 23:10:18.502 [scala-execution-context-global-10] ERROR com.github.gvolpe.http4s.server.endpoints.MultipartHttpEndpoint - HEADER 0: Content-Disposition: form-data; name="names_toto"; filename="geoflram-benchs.json"
root 23:10:18.502 [scala-execution-context-global-10] ERROR com.github.gvolpe.http4s.server.endpoints.MultipartHttpEndpoint - HUUUUUUUUUUUUUGE 11
root 23:10:18.525 [scala-execution-context-global-10] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Websocket key: None
root Request headers: Headers(Host: localhost:8080, User-Agent: curl/7.54.0, Accept: */*, Content-Length: 9823, Expect: 100-continue, Content-Type: multipart/form-data; boundary=------------------------39aaf72dd199bf5e)
root 23:10:18.743 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - closeConnection()
root 23:10:18.743 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Shutting down HttpPipeline
root 23:10:18.745 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Canceled request
root 23:10:18.745 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Shutting down.
root 23:10:18.745 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Shutting down HttpPipeline
root 23:10:18.746 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Canceled request
root 23:10:18.746 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Shutting down.
root 23:10:18.746 [blaze-selector-0-1] DEBUG org.http4s.blazecore.IdleTimeoutStage - Shutting down idle timeout stage
root 23:10:18.746 [blaze-selector-0-1] DEBUG org.http4s.blazecore.IdleTimeoutStage - Shutting down.
root 23:10:18.746 [blaze-selector-0-1] DEBUG org.http4s.blaze.channel.nio1.NIO1HeadStage - Shutting down.
root 23:10:18.749 [blaze-selector-0-1] DEBUG org.http4s.blaze.channel.nio1.NIO1HeadStage - Stage NIO1HeadStage sending inbound command: Disconnected
root 23:10:18.749 [blaze-selector-0-1] DEBUG org.http4s.blazecore.IdleTimeoutStage - Shutting down idle timeout stage
root 23:10:18.749 [blaze-selector-0-1] DEBUG org.http4s.blazecore.IdleTimeoutStage - Shutting down.
root 23:10:18.749 [blaze-selector-0-1] DEBUG org.http4s.blazecore.IdleTimeoutStage - Stage IdleTimeoutStage sending inbound command: Disconnected
root 23:10:18.749 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Shutting down HttpPipeline
root 23:10:18.749 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Canceled request
root 23:10:18.749 [blaze-selector-0-1] DEBUG org.http4s.server.blaze.Http1ServerStage$$anon$1 - Shutting down.
```

Reply:
Have you tried passing in a Blocking Execution Context? You shouln't use `global` here. Many things have changed in a year!